### PR TITLE
add input option controlling if negative weights are allowed in IBFE coupling

### PR DESCRIPTION
**Pull request by boyceg**
This option should have been included in #1188.

### IBAMR Pull Request Checklist
- [x] Does the test suite pass?
- [x] Was clang-format run?
- [x] Were relevant issues cited? Please remember to add `Fixes #12345` to close
      the issue automatically if we are fixing the problem.
- [x] Is this a change others will want to know about? If so, then has a
      changelog entry been added?
- [x] If new data structures have been added to a class then have they been set
      up appropriately for restarts? If so, ensure that the restart version
      number is incremented.
- [x] Does this change include a bug fix or new functionality? If so, a new test
      or tests should be added. New tests should run quickly (less than a minute
      in release mode). If possible, an older test should gain a new option so
      that we do not need to compile more test executables.
- [x] Did you (if your account has permission to do so) set relevant labels on
      GitHub for the pull request?

### CODE REVIEW

File: src/IB/IBFEMethod.cpp [hunk 1]
```diff
@@ -2966,6 +2966,7 @@ IBFEMethod::getFromInput(const Pointer<Database>& db, bool /*is_from_restart*/)
         d_default_interp_spec.quad_order = FIRST;
         d_default_interp_spec.use_adaptive_quadrature = false;
         d_default_interp_spec.use_consistent_mass_matrix = false;
+        d_default_interp_spec.allow_rules_with_negative_weights = false;
```
Comment (drwells) [inline src/IB/IBFEMethod.cpp:2969]: Since this changes the default behavior I would prefer to wait on merging this until after the release.

Comment (boyceg) [reply src/IB/IBFEMethod.cpp:2969]: It does not change the default behavior because the nodal rules only have positive weights.

Comment (drwells) [reply src/IB/IBFEMethod.cpp:2969]: oh, I see - this is for `nodal_quadrature`. I should have scrolled up.

Comment (boyceg) [reply src/IB/IBFEMethod.cpp:2969]: Most of these settings do not have any impact for the specialized nodal interaction code path.

**Review by drwells — APPROVED** [review]

**Pull request merged**